src/main/java/org/terasology/scenario/components/ScenarioComponent.java

```diff
@@ -17,14 +17,46 @@
 
 import org.terasology.entitySystem.Component;
 import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.math.Region3i;
 import org.terasology.network.FieldReplicateType;
 import org.terasology.network.Replicate;
+import org.terasology.scenario.internal.systems.RegionSystem;
+import org.terasology.structureTemplates.components.ProtectedRegionsComponent;
 
 import java.util.ArrayList;
 import java.util.List;
 
 /**
- * Component attached to a scenario entity. Denotes the "root" of the logic structure.
+ *  Component that indicates an entity is the root scenario entity. Only one entity with this component should exist at a time.
+ *  Trigger entities list contain a list of triggers that each have a list of logic entities
+ *  Region entities list contains a list of all the region entities
+ *
+ * Argument entities include:
+ *   Network Component
+ *   Type Component
+ *   Value or Expression Component (Values are constant values, expressions are evaluated to obtain the value)
+ *
+ * Typical Scenario logic entities include:
+ *   Network Component
+ *   Indicator Component
+ *   {@link ScenarioLogicLabelComponent} - label field includes the text for the dropdown menus
+ *   Secondary Component
+ *   *{@link ScenarioLogicTextComponent} - text field is the text that is displayed with arguments included (Detailed in the class)
+ *   *{@link ScenarioArgumentContainerComponent} - Only needed if the entity description includes argument parameters
+ *

```

Comment:
This is redundant to the javadoc in the class above. Please just mention in the javadoc that it is explained elsewhere. Otherwise when it changes multiple places need to be changed. It is as bad as duplicate code.